geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/tube/TubeBuilder.scala

```diff
@@ -97,6 +97,55 @@ abstract class TubeBuilder(val tubeFeatures: SimpleFeatureCollection,
     }
   }
 
+  /**
+    * This function checks if a segment crosses the IDL.
+    * @param point1 The first point in the segment
+    * @param point2 The second point in the segment
+    * @return boolean true if the segment crosses the IDL, otherwise false
+    */
+  def crossesIDL(point1:Coordinate,point2:Coordinate): Boolean = {
+    ((point1.x > 0 && point2.x < 0) || (point2.x > 0 && point1.x < 0))
+  }
+
+  /**
+    * Calculate the latitude at which the segment intercepts the IDL.
+    * @param point1 The first point in the segment
+    * @param point2 The second point in the segment
+    * @return a dobule representing the intercept latitude
+    */
+  def calcIDLIntercept(point1: Coordinate, point2: Coordinate): Double = {
+    if(point2.x > point1.x){
+      point2.y - (((point2.y-point1.y)/(point2.x-(-180+point1.x)))*point2.x)
```

Comment:
Okay, I'm probably just doing the algebra wrong, but when I put in the proportions, I don't get this formula.  Instead, I get something like:  `intercept = y2 - (y2 - y1)(x2 - 180)/(x2 - x1)`  Clearly, using 180 as the X-value where the segment between the two points intercepts the anti-meridian.  Please let me know what I've mistaken.  Thanks!